### PR TITLE
Replace %p with %s for strings in debug

### PR DESCRIPTION
**Pull request by AtariDreams**
Debugging strings makes ore sense if we get their string value instead of just their address. So most of these changes should be correct.

### CODE REVIEW

File: backend/ipp.c [hunk 1]
```diff
@@ -2931,9 +2931,9 @@ password_cb(const char *prompt,		/* I - Prompt (not used) */
 
 
   fprintf(stderr, "DEBUG: password_cb(prompt=\"%s\", http=%p, method=\"%s\", "
-                  "resource=\"%s\", password_tries=%p(%d)), password=%p\n",
+          "resource=\"%s\", password_tries=%p(%d)), password=%p\n",
```
Comment (michaelrsweet) [inline backend/ipp.c:2934]: I'm not keen on losing the indentation to match up the starts of each piece of the format string. Either preserve the indentation or eliminate the newlines entirely (current coding style no longer requires manual wrapping at 80 columns...)

File: cups/ppd-util.c [hunk 2]
```diff
@@ -145,7 +145,7 @@ cupsGetPPD3(http_t     *http,		/* I  - HTTP connection or @code CUPS_HTTP_DEFAUL
   * Range check input...
   */
 
-  DEBUG_printf(("cupsGetPPD3(http=%p, name=\"%s\", modtime=%p(%d), buffer=%p, "
+  DEBUG_printf(("cupsGetPPD3(http=%p, name=\"%s\", modtime=%p(%d), buffer=%s, "
```
Comment (michaelrsweet) [inline cups/ppd-util.c:148]: This buffer may not be initialized, so it probably isn't safe to use %s here...

File: cups/ppd-util.c [hunk 3]
```diff
@@ -562,7 +562,7 @@ cups_get_printer_uri(
 		};
 
 
-  DEBUG_printf(("4cups_get_printer_uri(http=%p, name=\"%s\", host=%p, hostsize=%d, resource=%p, resourcesize=%d, depth=%d)", http, name, host, hostsize, resource, resourcesize, depth));
+  DEBUG_printf(("4cups_get_printer_uri(http=%p, name=\"%s\", host=%s, hostsize=%d, resource=%s, resourcesize=%d, depth=%d)", http, name, host, hostsize, resource, resourcesize, depth));
```
Comment (michaelrsweet) [inline cups/ppd-util.c:565]: Again, host isn't initialized until this function returns...

File: cups/ppd.c [hunk 4]
```diff
@@ -611,7 +611,7 @@ _ppdOpen(
     return (NULL);
   }
 
-  DEBUG_printf(("2_ppdOpen: keyword=%s, string=%p", keyword, string));
+  DEBUG_printf(("2_ppdOpen: keyword=%s, string=%s", keyword, string));
```
Comment (michaelrsweet) [inline cups/ppd.c:614]: The string buffer may not be initialized at this point...

File: cups/sidechannel.c [hunk 5]
```diff
@@ -110,7 +110,7 @@ cupsSideChannelRead(
 #endif /* HAVE_POLL */
 
 
-  DEBUG_printf(("cupsSideChannelRead(command=%p, status=%p, data=%p, "
+  DEBUG_printf(("cupsSideChannelRead(command=%p, status=%p, data=%s, "
```
Comment (michaelrsweet) [inline cups/sidechannel.c:113]: Side-channel command data is binary so we can't use %s here.

File: cups/sidechannel.c [hunk 6]
```diff
@@ -304,7 +304,7 @@ cupsSideChannelSNMPGet(
 			real_oidlen;	/* Length of returned OID string */
 
 
-  DEBUG_printf(("cupsSideChannelSNMPGet(oid=\"%s\", data=%p, datalen=%p(%d), "
+  DEBUG_printf(("cupsSideChannelSNMPGet(oid=\"%s\", data=%s, datalen=%p(%d), "
```
Comment (michaelrsweet) [inline cups/sidechannel.c:307]: Ditto, binary data.

File: cups/snmp.c [hunk 7]
```diff
@@ -261,7 +261,7 @@ _cupsSNMPOIDToString(const int *src,	/* I - OID */
 	*dstend;			/* End of string buffer */
 
 
-  DEBUG_printf(("4_cupsSNMPOIDToString(src=%p, dst=%p, dstsize=" CUPS_LLFMT ")",
+  DEBUG_printf(("4_cupsSNMPOIDToString(src=%p, dst=%s, dstsize=" CUPS_LLFMT ")",
```
Comment (michaelrsweet) [inline cups/snmp.c:264]: "dst" is not yet initialized.

File: scheduler/client.c [hunk 8]
```diff
@@ -2978,7 +2978,7 @@ get_file(cupsd_client_t *con,		/* I  - Client connection */
     }
   }
 
-  cupsdLogClient(con, CUPSD_LOG_DEBUG2, "get_file: filestats=%p, filename=%p, len=" CUPS_LLFMT ", returning \"%s\".", filestats, filename, CUPS_LLCAST len, status ? "(null)" : filename);
+  cupsdLogClient(con, CUPSD_LOG_DEBUG2, "get_file: filestats=%p, filename=%s, len=" CUPS_LLFMT ", returning \"%s\".", filestats, filename, CUPS_LLCAST len, status ? "(null)" : filename);
```
Comment (michaelrsweet) [inline scheduler/client.c:2981]: Here we specifically report both the address *and* the value (if initialized), so no change here please.

File: scheduler/process.c [hunk 9]
```diff
@@ -445,7 +445,7 @@ cupsdFinishProcess(int    pid,		/* I - Process ID */
     strlcpy(name, "unknown", namelen);
   }
 
-  cupsdLogMessage(CUPSD_LOG_DEBUG2, "cupsdFinishProcess(pid=%d, name=%p, namelen=" CUPS_LLFMT ", job_id=%p(%d)) = \"%s\"", pid, name, CUPS_LLCAST namelen, job_id, job_id ? *job_id : 0, name);
+  cupsdLogMessage(CUPSD_LOG_DEBUG2, "cupsdFinishProcess(pid=%d, name=%s, namelen=" CUPS_LLFMT ", job_id=%p(%d)) = \"%s\"", pid, name, CUPS_LLCAST namelen, job_id, job_id ? *job_id : 0, name);
```
Comment (michaelrsweet) [inline scheduler/process.c:448]: Here again we show both the address and the value, so no change...

**Review by michaelrsweet — CHANGES_REQUESTED** [review]
Aside from adding the `void *` cast, I don't think we want any of these changes as-is.

**Comment by AtariDreams** [comment]
So you just want the address in debug, not the values? 

Just asking

**Pull request closed without merging**